controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -225,7 +225,7 @@ func (r *HostedControlPlaneReconciler) buildInternalMonitoringRouteMonitor(route
 			Route: v1alpha1.RouteMonitorRouteSpec{
 				Name:      route.Name,
 				Namespace: route.Namespace,
-				Port:      6443,
+				Port:      int64(hostedcontrolplane.Status.ControlPlaneEndpoint.Port),
```

Comment:
Is this is problematic if the status field isn't populated yet (since status fields aren't guaranteed to be populated)?